src/modules/producer.js

```diff
@@ -118,7 +119,7 @@ function checkRpc (queue, msg, options) {
 
         if (publishOrSendToQueue.call(this, queue, msg, options)) {
             //defered promise that will resolve when response is received
-            let responsePromise = Promise.defer();
+            let responsePromise =  new Deferred();
```

Comment:
OK I understand if this is a quick fix then it is fine there is no simple replacement so yours is ok. however this whole thing should be refactored to not use a defer mechanism if possible (i did not go through the code in detail to know exactly which way to do it now)  at all. It is a little bigger refactor for sure.